frontend/src/components/Admin/EditTeam.tsx

```diff
@@ -219,13 +219,17 @@ class EditTeam extends React.Component<Record<string, unknown>, EditTeamState> {
                         matchChecker={(query: string, member: Member) => {
                           const queryLower = query.toLowerCase();
                           return (
-                            member.email.toLowerCase().startsWith(queryLower) ||
-                            member.firstName.toLowerCase().startsWith(queryLower) ||
-                            member.lastName.toLowerCase().startsWith(queryLower) ||
-                            member.role.toLowerCase().startsWith(queryLower) ||
-                            `${member.firstName.toLowerCase()} ${member.lastName.toLowerCase()}`.startsWith(
-                              queryLower
-                            )
+                            (member.email && member.email.toLowerCase().startsWith(queryLower)) ||
+                            (member.firstName &&
+                              member.firstName.toLowerCase().startsWith(queryLower)) ||
+                            (member.lastName &&
+                              member.lastName.toLowerCase().startsWith(queryLower)) ||
+                            (member.role && member.role.toLowerCase().startsWith(queryLower)) ||
+                            ((member.firstName &&
+                              member.lastName &&
+                              `${member.firstName.toLowerCase()} ${member.lastName.toLowerCase()}`.startsWith(
+                                queryLower
+                              )) as boolean)
```

Comment:
While you are here, let's replace the copy-pasta with one common checker.

Reply:
Was running into some issues while trying to refactor the code. Can I make a separate PR for refactoring the code?